File: api/history/snapshot/[snapshot_id].py
"""
Vercel Serverless Function: GET /api/history/snapshot/[snapshot_id]
"""
import sys
import os
import json

project_root = os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))))
sys.path.insert(0, project_root)

from http.server import BaseHTTPRequestHandler

class handler(BaseHTTPRequestHandler):
    def do_GET(self):
        import asyncio
        from database import get_snapshot_data
        
        # 경로에서 snapshot_id 추출
        path = self.path.strip('/')
        path_parts = path.split('/')
        
        snapshot_id = None
        if len(path_parts) >= 4 and path_parts[0] == 'api' and path_parts[1] == 'history' and path_parts[2] == 'snapshot':
            try:
                snapshot_id = int(path_parts[3])
            except ValueError:
                pass
        
        if not snapshot_id:
            self.send_response(400)
            self.send_header('Content-Type', 'application/json; charset=utf-8')
            self.end_headers()
            response = json.dumps({"error": "snapshot_id is required"}, ensure_ascii=False)
            self.wfile.write(response.encode('utf-8'))
            return
        
        try:
            data = asyncio.run(get_snapshot_data(snapshot_id))
            
            self.send_response(200)
            self.send_header('Content-Type', 'application/json; charset=utf-8')
            self.send_header('Access-Control-Allow-Origin', '*')
            self.end_headers()
            
            response = json.dumps(data, ensure_ascii=False)
            self.wfile.write(response.encode('utf-8'))
            
        except Exception as e:
            self.send_response(200)
            self.send_header('Content-Type', 'application/json; charset=utf-8')
            self.send_header('Access-Control-Allow-Origin', '*')
            self.end_headers()
            
            response = json.dumps({}, ensure_ascii=False)
            self.wfile.write(response.encode('utf-8'))
    
    def do_OPTIONS(self):
        self.send_response(200)
        self.send_header('Access-Control-Allow-Origin', '*')
        self.end_headers()

File: api/history/snapshot/[snapshot_id].js
import { createClient } from '@supabase/supabase-js';

export const config = { runtime: 'nodejs' };

export default async function handler(req, res) {
  res.setHeader('Access-Control-Allow-Origin', '*');
  if (req.method === 'OPTIONS') {
    res.status(200).end();
    return;
  }
  if (req.method !== 'GET') {
    res.status(405).json({ error: 'Method not allowed' });
    return;
  }

  // Vercel: 동적 경로 /api/history/snapshot/123 → query 또는 URL에서 추출
  let snapshot_id = req.query.snapshot_id;
  if (!snapshot_id && req.url) {
    const match = req.url.match(/\/snapshot\/([^/?#]+)/);
    if (match) snapshot_id = match[1];
  }
  if (!snapshot_id) {
    return res.status(400).json({ error: 'snapshot_id required' });
  }

  const url = process.env.SUPABASE_URL;
  const key = process.env.SUPABASE_SERVICE_ROLE_KEY || process.env.SUPABASE_ANON_KEY;
  if (!url || !key) {
    return res.status(200).json([]);
  }

  try {
    const supabase = createClient(url, key);
    const { data: rows, error } = await supabase
      .from('weather_data')
      .select('data_json')
      .eq('snapshot_id', Number(snapshot_id))
      .order('airport_name');
    if (error) return res.status(200).json([]);
    const list = (rows || []).map((r) => {
      try {
        return JSON.parse(r.data_json);
      } catch {
        return null;
      }
    }).filter(Boolean);
    return res.status(200).json(list);
  } catch (e) {
    return res.status(200).json([]);
  }
}
